src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,255 +42,151 @@
-        if ((bytesRead == 0) && (dataSize == 0))
-            break;
+        lt::address_v4::bytes_type octets = {};
+        unsigned char octetIndex = 0;
 
-        for (start = 0; start < dataSize; ++start)
+        for (const char *lastptr = IPstr.data();
+             (lastptr < (IPstr.data() + IPstr.size())) && (octetIndex < 4)
+             ;)
         {
-            endOfLine = -1;
-            // The file might have ended without the last line having a newline
-            if (!((bytesRead == 0) && (dataSize > 0)))
-            {
-                for (int i = start; i < dataSize; ++i)
-                {
-                    if (buffer[i] == '\n')
-                    {
-                        endOfLine = i;
-                        // We need to NULL the newline in case the line has only an IP range.
-                        // In that case the parser won't work for the end IP, because it ends
-                        // with the newline and not with a number.
-                        buffer[i] = '\0';
-                        break;
-                    }
-                }
-            }
-            else
-            {
-                endOfLine = dataSize;
-                buffer[dataSize] = '\0';
-            }
+            lt::address_v4::bytes_type::value_type octet = 0;
 
-            if (endOfLine == -1)
-            {
-                // read the next chunk from file
-                // but first move(copy) the leftover data to the front of the buffer
-                offset = dataSize - start;
-                memmove(buffer.data(), buffer.data() + start, offset);
-                break;
-            }
+            auto posIt  = std::find_if_not(lastptr, IPstr.data() + IPstr.size(),
+                                          [](unsigned char c){ return std::isspace(c); } );
+            int pos = 0;
+            if (posIt != (IPstr.data() + IPstr.size()))
+                pos = posIt - lastptr;
 
-            ++nbLine;
+            const auto &[ptr, ec] = std::from_chars((lastptr + pos), (IPstr.data() + IPstr.size()), octet);
+            if (ec != std::errc())
+                return {};
 
-            if ((buffer[start] == '#')
-                || ((buffer[start] == '/') && ((start + 1 < dataSize) && (buffer[start + 1] == '/'))))
-                {
-                start = endOfLine;
-                continue;
-            }
+            octets[octetIndex++] = octet;
+            lastptr = ptr + 1;
+        }
 
-            // Each line should follow this format:
-            // 001.009.096.105 - 001.009.096.105 , 000 , Some organization
-            // The 3rd entry is access level and if above 127 the IP range isn't blocked.
-            const int firstComma = findAndNullDelimiter(buffer.data(), ',', start, endOfLine);
-            if (firstComma != -1)
-                findAndNullDelimiter(buffer.data(), ',', firstComma + 1, endOfLine);
+        if (octetIndex == 4)
+            return lt::address_v4(octets);
```

Comment:
Why not put it in loop body? Then you can omit additional cheking in loop condition.